docs/source/tutorials/qat_finetune.rst

```diff
@@ -0,0 +1,371 @@
+  for more details on how you can easily clone and modify torchtune configs.
+
+Quantizing the QAT model
+------------------------
+
+Note that the QAT recipe above produces an unquantized bfloat16 model. The model
+structure is exactly the same as the model produced with regular full fine-tuning
+without QAT, just with different weights. To actually get a quantized model,
+copy and make the following modifications to the quantization config:
+
+.. code-block:: bash
+
+  tune cp quantization custom_quantization.yaml
+
+.. code-block:: yaml
+
+  # Model arguments
+  model:
+    _component_: torchtune.models.llama3.llama3_8b
+
+  checkpointer:
+    _component_: torchtune.utils.FullModelMetaCheckpointer
+    checkpoint_dir: <your QAT checkpoint dir>
+    checkpoint_files: [meta_model_0.pt]
+    recipe_checkpoint: null
+    output_dir: <your QAT checkpoint dir>
+    model_type: LLAMA3
+
+  ...
+
+  quantizer:
+    _component_: torchtune.utils.quantization.Int8DynActInt4WeightQATQuantizer
+    groupsize: 256
+
+The following command performs the convert step in the QAT flow, which actually
+quantizes the float model to a model with quantized weights:
+
+.. code-block:: bash
+
+  tune run quantize --config custom_quantization.yaml
+
+.. note::
+
+  Make sure to use the same QAT quantizer you used to fine-tune your model,
+  otherwise the numerics will be off and the quantized model will perform poorly.
+
+Evaluating the quantized model
+------------------------------
+
+Now that we have a quantized model, we can run some evaluations on it and compare the
+results against regular fine-tuning without QAT (i.e. post-training quantization).
+To achieve this, we use `EleutherAI’s evaluation harness <https://github.com/EleutherAI/lm-evaluation-harness>`_
+integrated in torchtune. First, copy the evaluation config and make the following changes:
+
+.. code-block:: bash
+
+  tune cp eleuther_evaluation custom_eleuther_evaluation.yaml
+
+.. code-block:: yaml
+
+  # Model arguments
+  model:
+    _component_: torchtune.models.llama3.llama3_8b
+
+  checkpointer:
+    _component_: torchtune.utils.FullModelTorchTuneCheckpointer
+    checkpoint_dir: <your quantized model checkpoint dir>
+    checkpoint_files: [meta_model_0-8da4w.pt]
+    recipe_checkpoint: null
+    output_dir: <your quantized model checkpoint dir>
+    model_type: LLAMA3
+
+  ...
+
+  # EleutherAI specific eval args
+  tasks: ["hellaswag", "wikitext"]
+  limit: null
+  max_seq_length: 8192
+  batch_size: 8
+
+  quantizer:
+    _component_: torchtune.utils.quantization.Int8DynActInt4WeightQuantizer
+    groupsize: 256
+
+.. note::
+
+  Since we are passing in a quantized model, be sure to use the corresponding
+  post-training quantizer instead of the QAT quantizer. For example, if you
+  used the :code:`Int8DynActInt4WeightQATQuantizer` during fine-tuning, you
+  should specify :code:`Int8DynActInt4WeightQuantizer` in this step.
```

Comment:
Do we want to provide a reference somewhere to the currently-supported quantization methods?

Reply:
added a link